src/tim/layout_infer/layout_inference_test.cc

```diff
@@ -0,0 +1,60 @@
+#include "tim/layout_infer/layout_inference.h"
+#include "tim/vx/context.h"
+#include "tim/vx/graph.h"
+#include "tim/vx/ops/conv2d.h"
+
+#include "gtest/gtest.h"
+
+TEST(LayoutInference, simple_conv2d) {
+  auto ctx = tim::vx::Context::Create();
+  auto src_graph = ctx->CreateGraph();
+  tim::vx::ShapeType input_shape({1, 3, 3, 1});
+  tim::vx::TensorSpec input_spec(tim::vx::DataType::FLOAT32, input_shape,
+                                 tim::vx::TensorAttribute::INPUT);
+  auto input = src_graph->CreateTensor(input_spec);
+
+  tim::vx::ShapeType kernel_shape({1, 2, 2, 1});
+  tim::vx::TensorSpec kernel_spec(tim::vx::DataType::FLOAT32, kernel_shape,
+                                  tim::vx::TensorAttribute::CONSTANT);
+  std::vector<float> kernel_data = {
+      0.25f, 0.25f, 0.25f, 0.25f};
+  auto kernel = src_graph->CreateTensor(kernel_spec, kernel_data.data());
+
+  tim::vx::ShapeType bias_shape({1});
+  tim::vx::TensorSpec bias_spec(tim::vx::DataType::FLOAT32, bias_shape,
+                                tim::vx::TensorAttribute::CONSTANT);
+  std::vector<float> bias_data = {0.0f};
+  auto bias = src_graph->CreateTensor(bias_spec, bias_data.data());
+
+  tim::vx::ShapeType output_shape({1, 2, 2, 1});
+  tim::vx::TensorSpec output_spec(tim::vx::DataType::FLOAT32, output_shape,
+                                  tim::vx::TensorAttribute::OUTPUT);
+  auto output = src_graph->CreateTensor(output_spec);
+
+  auto conv2d = src_graph->CreateOperation<tim::vx::ops::Conv2d>(
+      kernel_shape[0], tim::vx::PadType::AUTO,
+      std::array<uint32_t, 2>({kernel_shape[2], kernel_shape[1]}),
+      std::array<uint32_t, 2>({1, 1}), std::array<uint32_t, 2>({0, 0}),
+      tim::vx::DataLayout::CWHN, std::array<uint32_t, 4>({0, 0, 0, 0}));
+  (*conv2d).BindInputs({input, kernel, bias}).BindOutput(output);
+
+  auto infer_graph = ctx->CreateGraph();
+  auto layout_infer = std::make_shared<tim::layout_inference::LayoutInfer>(src_graph, infer_graph);
```

Comment:
auto infer_graph = LayoutInfer(src_graph).run(); // create graph inside layoutinference

API usage change request.

Reply:
Ok. So we need transfer the ctx in LayoutInfer.